library_generation/owlbot/synthtool/languages/java.py

```diff
@@ -0,0 +1,691 @@
+    """
+    s.replace(
+        _filter_no_header([grpc_root / "src/**/*.java"]),
+        "^package (.*);",
+        f"{GOOD_LICENSE}package \\1;",
+    )
+
+
+def latest_maven_version(group_id: str, artifact_id: str) -> Optional[str]:
+    """Helper function to find the latest released version of a Maven artifact.
+
+    Fetches metadata from Maven Central and parses out the latest released
+    version.
+
+    Args:
+        group_id (str): The groupId of the Maven artifact
+        artifact_id (str): The artifactId of the Maven artifact
+
+    Returns:
+        The latest version of the artifact as a string or None
+    """
+    group_path = "/".join(group_id.split("."))
+    url = (
+        f"https://repo1.maven.org/maven2/{group_path}/{artifact_id}/maven-metadata.xml"
+    )
+    response = requests.get(url)
+    if response.status_code >= 400:
+        return "0.0.0"
+
+    return version_from_maven_metadata(response.text)
+
+
+def version_from_maven_metadata(metadata: str) -> Optional[str]:
+    """Helper function to parse the latest released version from the Maven
+    metadata XML file.
+
+    Args:
+        metadata (str): The XML contents of the Maven metadata file
+
+    Returns:
+        The latest version of the artifact as a string or None
+    """
+    root = ET.fromstring(metadata)
+    latest = root.find("./versioning/latest")
+    if latest is not None:
+        return latest.text
+
+    return None
+
+
+def _common_generation(
+    service: str,
+    version: str,
+    library: Path,
+    package_pattern: str,
+    suffix: str = "",
+    destination_name: str = None,
+    cloud_api: bool = True,
+    diregapic: bool = False,
+    preserve_gapic: bool = False,
+):
+    """Helper function to execution the common generation cleanup actions.
+
+    Fixes headers for protobuf classes and generated gRPC stub services. Copies
+    code and samples to their final destinations by convention. Runs the code
+    formatter on the generated code.
+
+    Args:
+        service (str): Name of the service.
+        version (str): Service API version.
+        library (Path): Path to the temp directory with the generated library.
+        package_pattern (str): Package name template for fixing file headers.
+        suffix (str, optional): Suffix that the generated library folder. The
+            artman output differs from bazel's output directory. Defaults to "".
+        destination_name (str, optional): Override the service name for the
+            destination of the output code. Defaults to the service name.
+        preserve_gapic (bool, optional): Whether to preserve the gapic directory
+            prefix. Default False.
+    """
+
+    if destination_name is None:
+        destination_name = service
+
+    cloud_prefix = "cloud-" if cloud_api else ""
+    package_name = package_pattern.format(service=service, version=version)
+    fix_proto_headers(
+        library / f"proto-google-{cloud_prefix}{service}-{version}{suffix}"
```

Comment:
nit: could we use a variable for `f"proto-google-{cloud_prefix}{service}-{version}{suffix}"` to reduce the length of the line.

Reply:
Yes, Done.